eclipse/src/saros/ui/util/CollaborationUtils.java

```diff
@@ -276,6 +266,32 @@ public void run() {
         });
   }
 
+  /**
+   * Returns the reference point objects representing the given container objects.
+   *
+   * @param referencePointContainers the container objects to represent as reference points
+   * @return the reference point objects representing the given container objects
+   */
+  private static Set<IReferencePoint> getReferencePoints(Set<IContainer> referencePointContainers) {
+    return referencePointContainers
+        .stream()
+        .map(EclipseReferencePointImpl::new)
+        .collect(Collectors.toSet());
+  }
+
+  /**
+   * Returns all projects that contain at least one of the given reference points.
+   *
+   * @param referencePoints the currently shared reference points
+   * @return all projects that contain at least one of the given reference points
+   */
+  private static Set<IProject> getProjects(Set<IReferencePoint> referencePoints) {
+    return referencePoints
+        .stream()
+        .map(referencePoint -> ResourceConverter.getDelegate(referencePoint).getProject())
```

Comment:
Is it save to call `getProject` or should we check for `null`?

Reply:
`null` could only be returned if the reference point would be the workspace root. If you would like to add a sanity check you have to check that no passed in IContainer is the workspace root.